components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -116,6 +119,12 @@ void ViewCounterService::BrandedWallpaperWillBeDisplayed(
         wallpaper_id ? *wallpaper_id : "",
         creative_instance_id ? *creative_instance_id : "",
         ads::mojom::NewTabPageAdEventType::kViewed);
+
+    if (ntp_p3a_helper_ && !ads_service_->IsEnabled()) {
+      // Should only report to P3A if ads are disabled, as required by spec.
+      ntp_p3a_helper_->RecordView(creative_instance_id ? *creative_instance_id
+                                                       : "");
+    }
```

Comment:
@aseren, this code should only be called when an NTP-SI has been viewed if a user has not joined rewards/ads are disabled. Can you please review this change? Thanks

Reply:
This should work fine